Existing github action ci is returning errors - missing packaging

https://github.com/opendatacube/odc-tools/actions

```
Run find . -name _version.py | xargs python ./scripts/patch_version.py ${GITHUB_RUN_NUMBER:-0}
Traceback (most recent call last):
  File "./scripts/patch_version.py", line 2, in <module>
    from packaging import version
ModuleNotFoundError: No module named 'packaging'
Error: Process completed with exit code 123.
```

and

```
=================================== ERRORS ====================================
_________ ERROR collecting apps/dc_tools/tests/test_stac_api_to_dc.py __________
apps/dc_tools/tests/test_stac_api_to_dc.py:4: in <module>
    from odc.apps.dc_tools.stac_api_to_dc import cli
tests/env/lib/python3.8/site-packages/odc/apps/dc_tools/stac_api_to_dc.py:22: in <module>
    from pystac_client import Client
tests/env/lib/python3.8/site-packages/pystac_client/__init__.py:3: in <module>
    from pystac_client.client import Client
tests/env/lib/python3.8/site-packages/pystac_client/client.py:8: in <module>
    from pystac_client.collection_client import CollectionClient
tests/env/lib/python3.8/site-packages/pystac_client/collection_client.py:7: in <module>
    from pystac_client.item_search import ItemSearch
tests/env/lib/python3.8/site-packages/pystac_client/item_search.py:17: in <module>
    from pystac_client.stac_api_io import StacApiIO
tests/env/lib/python3.8/site-packages/pystac_client/stac_api_io.py:[31](https://github.com/opendatacube/odc-tools/runs/6976564857?check_suite_focus=true#step:8:32): in <module>
    class StacApiIO(DefaultStacIO):
tests/env/lib/python3.8/site-packages/pystac_client/stac_api_io.py:57: in StacApiIO
    def read_text(self, source: pystac.link.HREF, *args: Any, **kwargs: Any) -> str:
E   AttributeError: module 'pystac.link' has no attribute 'HREF'
=============================== warnings summary ===============================
tests/env/lib/python3.8/site-packages/datacube/utils/geometry/_base.py:1[53](https://github.com/opendatacube/odc-tools/runs/6976564857?check_suite_focus=true#step:8:54)
tests/env/lib/python3.8/site-packages/datacube/utils/geometry/_base.py:153
  /home/runner/work/odc-tools/odc-tools/tests/env/lib/python3.8/site-packages/datacube/utils/geometry/_base.py:153: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    DEFAULT_WKT_VERSION = (WktVersion.WKT1_GDAL if LooseVersion(rasterio.__gdal_version__) < LooseVersion("3.0.0")
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR apps/dc_tools/tests/test_stac_api_to_dc.py - AttributeError: module 'py...
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
========================= 2 warnings, 1 error in 2.60s =========================
```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -152,8 +152,7 @@ jobs:
       shell: bash
       run: |
         datacube system check
-        pip freeze
-        pip list --format=freeze
+        
         echo "Running Tests"
         pytest --cov=. \
         --cov-report=html \

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -124,6 +124,7 @@ jobs:
       shell: bash
       run: |
         echo "$(pwd)/tests/env/bin" >> $GITHUB_PATH
+        export PATH="$(pwd)/tests/env/bin"
 
     - name: Install in Edit mode
       shell: bash
@@ -134,7 +135,6 @@ jobs:
 
         ./scripts/dev-install.sh --no-deps
 
-
     - name: Start Test DB
       shell: bash
       run: |
@@ -144,6 +144,7 @@ jobs:
         pg_ctl -D ${pgdata} -l "${pgdata}/pg.log" start
         createdb datacube
         datacube system init
+        pip list --format=freeze
 
       env:
         DATACUBE_DB_URL: postgresql:///datacube

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -72,7 +72,7 @@ jobs:
     - uses: conda-incubator/setup-miniconda@v2
       if: steps.conda_cache.outputs.cache-hit != 'true'
       with:
-        channels: conda-forge,defaults
+        channels: conda-forge,defaults,pystac
         channel-priority: true
         activate-environment: ""
         mamba-version: "*"
@@ -125,8 +125,6 @@ jobs:
       run: |
         echo "$(pwd)/tests/env/bin" >> $GITHUB_PATH
         export PATH="$(pwd)/tests/env/bin"
-        pip list --format=freeze
-
 
     - name: Install in Edit mode
       shell: bash

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -124,20 +124,18 @@ jobs:
       shell: bash
       run: |
         echo "$(pwd)/tests/env/bin" >> $GITHUB_PATH
+        export PATH="$(pwd)/tests/env/bin"
         pip list --format=freeze
+
 
     - name: Install in Edit mode
       shell: bash
       run: |
-        pip list --format=freeze
-
         which python
         which createdb
         which datacube
 
         ./scripts/dev-install.sh --no-deps
-
-        pip list --format=freeze
 
     - name: Start Test DB
       shell: bash

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -72,7 +72,7 @@ jobs:
     - uses: conda-incubator/setup-miniconda@v2
       if: steps.conda_cache.outputs.cache-hit != 'true'
       with:
-        channels: conda-forge,defaults,pystac
+        channels: conda-forge,defaults
         channel-priority: true
         activate-environment: ""
         mamba-version: "*"

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -123,7 +123,6 @@ jobs:
     - name: Update PATH
       shell: bash
       run: |
-        pip list --format=freeze
         echo "$(pwd)/tests/env/bin" >> $GITHUB_PATH
         pip list --format=freeze
 

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -125,6 +125,7 @@ jobs:
       run: |
         pip list --format=freeze
         echo "$(pwd)/tests/env/bin" >> $GITHUB_PATH
+        pip list --format=freeze
 
     - name: Install in Edit mode
       shell: bash

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -129,10 +129,11 @@ jobs:
     - name: Install in Edit mode
       shell: bash
       run: |
+        pip list --format=freeze
+
         which python
         which createdb
         which datacube
-        pip list --format=freeze
 
         ./scripts/dev-install.sh --no-deps
 

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -152,7 +152,7 @@ jobs:
       shell: bash
       run: |
         datacube system check
-
+        pip freeze
         echo "Running Tests"
         pytest --cov=. \
         --cov-report=html \

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -124,7 +124,6 @@ jobs:
       shell: bash
       run: |
         echo "$(pwd)/tests/env/bin" >> $GITHUB_PATH
-        export PATH="$(pwd)/tests/env/bin"
 
     - name: Install in Edit mode
       shell: bash
@@ -153,7 +152,7 @@ jobs:
       shell: bash
       run: |
         datacube system check
-        
+
         echo "Running Tests"
         pytest --cov=. \
         --cov-report=html \

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -153,6 +153,7 @@ jobs:
       run: |
         datacube system check
         pip freeze
+        pip list --format=freeze
         echo "Running Tests"
         pytest --cov=. \
         --cov-report=html \

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -123,6 +123,7 @@ jobs:
     - name: Update PATH
       shell: bash
       run: |
+        pip list --format=freeze
         echo "$(pwd)/tests/env/bin" >> $GITHUB_PATH
 
     - name: Install in Edit mode
@@ -131,9 +132,11 @@ jobs:
         which python
         which createdb
         which datacube
+        pip list --format=freeze
 
         ./scripts/dev-install.sh --no-deps
 
+        pip list --format=freeze
 
     - name: Start Test DB
       shell: bash
@@ -144,6 +147,7 @@ jobs:
         pg_ctl -D ${pgdata} -l "${pgdata}/pg.log" start
         createdb datacube
         datacube system init
+        pip list --format=freeze
 
       env:
         DATACUBE_DB_URL: postgresql:///datacube
@@ -152,7 +156,7 @@ jobs:
       shell: bash
       run: |
         datacube system check
-        
+
         echo "Running Tests"
         pytest --cov=. \
         --cov-report=html \

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -35,6 +35,7 @@ jobs:
         python -m pip install --upgrade \
          toml \
          wheel \
+         packaging \
          twine
         python -m pip freeze
 

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -124,6 +124,7 @@ jobs:
       shell: bash
       run: |
         echo "$(pwd)/tests/env/bin" >> $GITHUB_PATH
+        export PATH="$(pwd)/tests/env/bin"
 
     - name: Install in Edit mode
       shell: bash
@@ -134,7 +135,6 @@ jobs:
 
         ./scripts/dev-install.sh --no-deps
 
-
     - name: Start Test DB
       shell: bash
       run: |
@@ -144,6 +144,7 @@ jobs:
         pg_ctl -D ${pgdata} -l "${pgdata}/pg.log" start
         createdb datacube
         datacube system init
+        pip list --format=freeze
 
       env:
         DATACUBE_DB_URL: postgresql:///datacube
@@ -152,7 +153,7 @@ jobs:
       shell: bash
       run: |
         datacube system check
-
+        
         echo "Running Tests"
         pytest --cov=. \
         --cov-report=html \

--- a/tests/test-env-py38.yml
+++ b/tests/test-env-py38.yml
@@ -24,8 +24,8 @@ dependencies:
   - tqdm
 
   # odc-apps-dc-tools
-  - pystac>=1.1.0
-  - pystac-client>=0.2.0
+  - pystac>=1.2.0
+  - pystac-client>=0.4.0
   - azure-storage-blob
   - fsspec
   - lxml # needed for thredds-crawler

--- a/tests/test-env-py38.yml
+++ b/tests/test-env-py38.yml
@@ -5,6 +5,7 @@
 name: odc-tests-py38
 channels:
   - conda-forge
+
 dependencies:
   - python=3.8
 
@@ -60,7 +61,7 @@ dependencies:
   - pip:
       # odc.apps.dc-tools
       - thredds-crawler
-      - rio-stac==0.3.0
+      - rio-stac
 
       # odc.algo optional dependency
       - hdstats

--- a/tests/test-env-py38.yml
+++ b/tests/test-env-py38.yml
@@ -5,6 +5,7 @@
 name: odc-tests-py38
 channels:
   - conda-forge
+
 dependencies:
   - python=3.8
 
@@ -24,8 +25,8 @@ dependencies:
   - tqdm
 
   # odc-apps-dc-tools
-  - pystac>=1.1.0
-  - pystac-client>=0.2.0
+  - pystac>=1.2.0
+  - pystac-client>=0.4.0
   - azure-storage-blob
   - fsspec
   - lxml # needed for thredds-crawler
@@ -60,7 +61,7 @@ dependencies:
   - pip:
       # odc.apps.dc-tools
       - thredds-crawler
-      - rio-stac==0.3.0
+      - rio-stac
 
       # odc.algo optional dependency
       - hdstats

